core/src/lib.rs

```diff
@@ -363,4 +375,39 @@ mod tests {
             ),
         ]
     }
+
+    // tests the bucket method of Interval
+    #[test]
+    fn test_bucket() {
+        let data = vec![
+            TestData::new(Interval::new(1.0, 10.0, 5).unwrap(), 8.0, Some(4)),
+            TestData::new(Interval::new(30.0, 100.0, 10).unwrap(), 1000.0, None),
+            TestData::new(Interval::new(30.0, 100.0, 10).unwrap(), 0.0, None),
+            TestData::new(Interval::new(30.0, 100.0, 10).unwrap(), 10.0 * 10.0, None),
+            TestData::new(Interval::new(-100.0, 100.0, 10).unwrap(), 0.0, Some(8)),
+            TestData::new(Interval::new(-100.0, 100.0, 10).unwrap(), -100.0, Some(0)),
+        ];
+
+        for test in data {
+            let actual = Interval::bucket(&test.interval, test.number);
+
+            assert_eq!(test.expected, actual)
+        }
+    }
+
+    struct TestData {
+        interval: Interval,
+        number: f64,
+        expected: Option<usize>,
+    }
+
+    impl TestData {
+        fn new(interval: Interval, number: f64, expected: Option<usize>) -> Self {
+            Self {
+                interval,
+                number,
+                expected,
+            }
+        }
+    }
```

Comment:
I don't like how generically named this struct is. I see two options:

1. Rename it to something like `BucketTest`
2. Move the `struct` definitions inside the bucket unit test